packages/@uppy/provider-views/src/ProviderView/ProviderView.jsx

```diff
@@ -263,28 +265,34 @@ export default class ProviderView extends View {
       const messages = []
       const newFiles = []
 
-      // eslint-disable-next-line no-unreachable-loop
       for (const file of currentSelection) {
         if (file.isFolder) {
           const { requestPath, name } = file
           let isEmpty = true
           let numNewFiles = 0
 
-          for await (const fileInFolder of this.recursivelyListAllFiles(requestPath)) {
-            const tagFile = this.getTagFile(fileInFolder)
-            const id = getSafeFileId(tagFile)
-            // If the same folder is added again, we don't want to send
-            // X amount of duplicate file notifications, we want to say
-            // the folder was already added. This checks if all files are duplicate,
-            // if that's the case, we don't add the files.
-            if (!this.plugin.uppy.checkIfFileAlreadyExists(id)) {
-              newFiles.push(fileInFolder)
-              numNewFiles++
-              this.setLoading(this.plugin.uppy.i18n('addedNumFiles', { numFiles: numNewFiles }))
+          const queue = new PQueue({ concurrency: 6 })
```

Comment:
You're creating the queue inside a loop? Probably better to initialise it on the class.

Reply:
I think I misunderstood, the queue is only created for 'files' that are folders, not all files. Seems okay then :)